File: repos_estudiantes_ALVAREZ_segunda_entrega/ILLANES_VASQUEZ_GABRIEL/Practica2prog/Composicion y Agregacion/Ejer3/Avion.java

package Compo3;

public class Avion {
    private String modelo, fabricante;
    private Parte[] partes;
    private int cantidad;

    public Avion(String modelo, String fabricante, int cantidad) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.cantidad = 0;
        this.partes= new Parte [cantidad];
    }
    
    public void agregarParte(Parte p){
        if (cantidad< partes.length){
            partes[cantidad++]=p;
            
        }
        
    }
    
    public void mostrar(){
        
        System.out.println("el modelo " +modelo+ " fabricante " +fabricante);
        for (int i=0; i<cantidad;i++){
                partes[i].mostrarInfo();
        }
    }
    
    

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

  
    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }
    
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/Chino_Blanco_Noemi/2Composicion/Ejercicio2p/Avion.py


class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso

    def mostrar_info(self):
        return f"Parte: {self.nombre}, Peso: {self.peso} kg"

class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []

    def agregar_parte(self, parte):
        self.partes.append(parte)

    def mostrar_avion(self):
        print(f" Modelo: {self.modelo}")
        print(f" Fabricante: {self.fabricante}")
        print(" Partes:")
        for parte in self.partes:
            print("  ", parte.mostrar_info())


print("Noemí Chino Blanco")

avion = Avion("Amazonas A256", "pando")
avion.agregar_parte(Parte("Motor", 1300))
avion.agregar_parte(Parte("Alas", 700))
avion.agregar_parte(Parte("Tren de aterrizaje", 400))

avion.mostrar_avion()
